src/funcx_common/response_errors/error_classes.py

```diff
@@ -163,6 +163,32 @@ def __init__(self, ep_uuid: t.Union[uuid.UUID, str]) -> None:
         self.uuid = ep_uuid
 
 
+class EndpointInUse(FuncxResponseError):
+    """Endpoint with specified uuid is already registered"""
+
+    code = ResponseErrorCode.RESOURCE_CONFLICT
+    http_status_code = HTTPStatusCode.RESOURCE_CONFLICT
+
+    def __init__(self, ep_uuid: t.Union[uuid.UUID, str]) -> None:
+        ep_uuid = str(ep_uuid)
+        self.error_args = [ep_uuid]
+        self.reason = f"Endpoint id {ep_uuid} is already registered to this user"
+        self.uuid = ep_uuid
+
+
+class EndpointLocked(FuncxResponseError):
+    """Endpoint with specified uuid cannot be registered until the lock expires"""
+
+    code = ResponseErrorCode.RESOURCE_LOCKED
+    http_status_code = HTTPStatusCode.RESOURCE_LOCKED
+
+    def __init__(self, ep_uuid: t.Union[uuid.UUID, str]) -> None:
+        ep_uuid = str(ep_uuid)
+        self.error_args = [ep_uuid]
+        self.reason = f"Endpoint id {ep_uuid} is temporarily locked from registration"
+        self.uuid = ep_uuid
```

Comment:
I don't yet know how this will be utilized, but I can see a potential use of knowing _when_ the lock will expire.  (For example, in a response to the end-user or in the logs.)

Reply:
I was debating whether to return the lock timestamp as part of the error, but felt it was extra logic (and passing of parameters) that may not be needed for a seconds long lock.  Could add it in, but let me think and we can discuss.

Reply:
lock timestamp added.